backend/app/services/knowledge/permission_request_service.py

```
@@ -0,0 +1,603 @@
+                Permission.kind_id == data.kind_id,
+                Permission.resource_type == "knowledge_base",
+                Permission.user_id == user_id,
+                Permission.status == PermissionStatus.PENDING,
+            )
+            .first()
+        )
+
+        if existing_request:
+            raise ValueError(
+                "You already have a pending request for this knowledge base"
+            )
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)